docs/operator-guides/scaling-airbyte.md

```diff
@@ -0,0 +1,72 @@
+
+These components perform control plane operations that are low-scale, low-resource work. In addition to the work being low cost, these components are efficient and optimized for these jobs, meaning that only uncommonly large workloads will require deployments at scale. In general, you would only encounter scaling issues when running over a thousand connections.
+
+As a reference point, the typical Airbyte user has 5 - 20 connectors and 10 - 100 connections configured. Almost all of these connections are scheduled,
+either hourly or daily, resulting in at most 100 concurrent jobs.
+
+## What To Scale
+[Workers](../understanding-airbyte/jobs.md) do all the heavy lifting within Airbyte. A worker is responsible for executing Airbyte operations (e.g. Discover, Read, Sync etc),
+and is created on demand whenever these operations are requested. Thus, every job has a corresponding worker executing its work.
+
+How a worker executes work depends on the Airbyte deployment. In the Docker deployment, an Airbyte worker spins up at least one Docker container. In the Kubernetes
+deployment, an Airbyte worker will create at least one Kubernetes pod. The created resource (Docker container or Kubernetes pod) does all the actual work.
+
+Thus, scaling Airbyte is a matter of ensuring that the Docker container or Kubernetes Pod running the jobs has sufficient resources to execute its work.
+
+Jobs-wise, we are mainly concerned with Sync jobs when thinking about scale. Sync jobs sync data from sources to destinations and are the majority of jobs run. Sync jobs use two workers.
+One worker reads from the source; the other worker writes to the destination.
+
+There are two resources to be aware of:
+1) Memory
+2) Disk space
+
+### Memory
+As mentioned above, we are mainly concerned with scaling Sync jobs. Within a Sync job, the main memory culprit is the Source worker.
+
+This is because the Source worker reads up to 10,000 records in memory. This can present problems for database sources with tables that have large row sizes. e.g. a table with an average row size of 0.5MBs will require 0.5 * 10000 / 1000 = 5GBs of RAM. See [this issue](https://github.com/airbytehq/airbyte/issues/3439) for more information.
+
+Our Java connectors currently follow Java's default behaviour with container memory and will only use up to 1/4 of the host's allocated memory. On Docker agent with 8GBs of RAM configured, a Java connector limits itself to 2Gbs of RAM and will see Out-of-Memory exceptions if this goes higher.
+
+Note that all Source database connectors are Java connectors. This means that users currently need to over-specify memory resource for Java connectors.
+
+On Docker, this can be solved by giving the Docker agent more memory. See [here](https://stackoverflow.com/questions/44533319/how-to-assign-more-memory-to-docker-container) for instructions. You might need to switch to a node with more memory. On Kubernetes, this can be solved by using a node type with more memory.
+
```

Comment:
We should add section here on resource defaults. I'm thinking we can add this after @ChristopheDuong writes up documentation explaining how to specify our resource requirements.

Although this isn't complete yet, thinking it's alright to merge in as is since it's better than what we have today.

Reply:
also happy to do this myself. regardless I think we can tackle this in a follow up PR.